x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/AutoFollowIT.java

```diff
@@ -471,81 +470,86 @@ public void testPauseAndResumeWithMultipleAutoFollowPatterns() throws Exception
             .build();
 
         final String[] prefixes = {"logs-", "users-", "docs-", "monitoring-", "data-", "system-", "events-", "files-"};
-        if (randomBoolean()) {
-            // sometimes create indices in the remote cluster that match the future auto follow patterns
-            Arrays.stream(prefixes).forEach(prefix -> createLeaderIndex(prefix + "ignored", leaderIndexSettings));
-        }
 
-        // create auto follow patterns
-        final List<String> autoFollowPatterns = new ArrayList<>(prefixes.length);
-        for (String prefix : prefixes) {
-            String name = prefix + "pattern";
-            putAutoFollowPatterns(name, new String[]{prefix + "*"});
-            autoFollowPatterns.add(name);
-            assertBusy(() -> assertThat(getAutoFollowStats().getAutoFollowedClusters().size(), equalTo(1)));
-            assertTrue(getAutoFollowPattern(name).isActive());
-        }
+        // create an auto follow pattern for each prefix
+        final List<String> autoFollowPatterns = Arrays.stream(prefixes)
+            .map(prefix -> {
+                final String pattern = prefix + "pattern";
+                putAutoFollowPatterns(pattern, new String[]{prefix + "*"});
+                return pattern;
+            }).collect(toUnmodifiableList());
 
-        // no following indices are created yet
-        assertThat(followerClient().admin().indices().prepareStats("copy-*").get().getIndices().size(), equalTo(0));
+        // pick up some random pattern to pause
+        final List<String> pausedAutoFollowerPatterns = randomSubsetOf(randomIntBetween(1, 3), autoFollowPatterns);
+
+        // all patterns should be active
+        assertBusy(() -> autoFollowPatterns.forEach(pattern -> assertTrue(getAutoFollowPattern(pattern).isActive())));
+        assertBusy(() -> assertThat(getAutoFollowStats().getAutoFollowedClusters().size(), equalTo(1)));
 
-        // create random indices in the remote cluster that match the patterns
         final AtomicBoolean running = new AtomicBoolean(true);
-        final Set<String> leaderIndices = ConcurrentCollections.newConcurrentSet();
+        final AtomicInteger leaderIndices = new AtomicInteger(0);
+
+        // start creating new indices on the remote cluster
         final Thread createNewLeaderIndicesThread = new Thread(() -> {
-            while (running.get()) {
+            while (running.get() && leaderIndices.get() < 20) {
+                final String prefix = randomFrom(prefixes);
+                final String leaderIndex = prefix + randomAlphaOfLength(10).toLowerCase(Locale.ROOT);
```

Comment:
I think we can have duplicate indices here.

Reply:
Right, thanks! I adapted the logic a bit to not create duplicate indices here.